django_loci/channels/base.py

```diff
@@ -29,12 +29,20 @@ def connect(self, message, pk):
 
     def is_authorized(self, user, location):
         perm = '{0}.change_location'.format(self.model._meta.app_label)
-        return user.is_authenticated() and (
-            user.is_superuser or (
-                user.is_staff and
-                user.has_perm(perm)
+        try:
+            return user.is_authenticated() and (
+                user.is_superuser or (
+                    user.is_staff and
+                    user.has_perm(perm)
+                )
+            )
+        except TypeError:
+            return user.is_authenticated and (
+                user.is_superuser or (
+                    user.is_staff and
+                    user.has_perm(perm)
+                )
```

Comment:
let's simplify this block of code and avoid duplication, try something like this:

```python
authenticated = user.is_authenticated
if callable(authenticated):
    authenticated = authenticated()
return authenticated and (
    # ... etcetera ...
)
```